homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,294 @@
+            ): bool,
+            vol.Optional(CONF_EVENT_CODE): str,
+            vol.Optional(CONF_DEVICE): vol.In(devices),
+            vol.Optional(CONF_REMOVE_DEVICE): vol.In(devices),
+        }
+
+        return self.async_show_form(
+            step_id="prompt_options", data_schema=vol.Schema(options), errors=errors
+        )
+
+    async def async_step_set_device_options(self, user_input=None):
+        """Manage device options."""
+        errors = {}
+
+        if user_input is not None:
+            device_id = get_device_id(
+                self._selected_device_object.device,
+                data_bits=user_input.get(CONF_DATA_BITS),
+            )
+            try:
+                command_on = (
+                    int(user_input.get(CONF_COMMAND_ON), 16)
+                    if user_input.get(CONF_COMMAND_ON)
+                    else None
+                )
+                command_off = (
+                    int(user_input.get(CONF_COMMAND_OFF), 16)
+                    if user_input.get(CONF_COMMAND_OFF)
+                    else None
+                )
+            except (NameError, ValueError):
+                errors = {"base": "invalid_input_2262"}
+
+            if not errors:
+                devices = {}
+                device = {
+                    CONF_DEVICE_ID: device_id,
+                    CONF_FIRE_EVENT: user_input.get(CONF_FIRE_EVENT, False),
+                    CONF_SIGNAL_REPETITIONS: user_input.get(CONF_SIGNAL_REPETITIONS, 1),
+                }
+
+                devices[self._selected_device_event_code] = device
+
+                if (
+                    user_input.get(CONF_OFF_DELAY_ENABLED)
+                    and user_input.get(CONF_OFF_DELAY) is not None
+                ):
+                    device[CONF_OFF_DELAY] = user_input[CONF_OFF_DELAY]
+                if user_input.get(CONF_DATA_BITS):
+                    device[CONF_DATA_BITS] = user_input.get(CONF_DATA_BITS)
+                if command_on:
+                    device[CONF_COMMAND_ON] = command_on
+                if command_off:
+                    device[CONF_COMMAND_OFF] = command_off
+
+                self.update_config_data(
+                    global_options=self._global_options, devices=devices
+                )
+
+                return self.async_create_entry(title="", data={})
+
+        device_data = self._selected_device
+
+        data_scheme = {
+            vol.Optional(
+                CONF_FIRE_EVENT, default=device_data.get(CONF_FIRE_EVENT, False)
+            ): bool,
+        }
+
+        if binary_supported(self._selected_device_object):
+            data_scheme.update(
+                {
+                    vol.Optional(
+                        CONF_OFF_DELAY_ENABLED,
+                        default=device_data.get(CONF_OFF_DELAY) is not None,
+                    ): bool,
+                    vol.Optional(
+                        CONF_OFF_DELAY, default=device_data.get(CONF_OFF_DELAY, 0)
```

Comment:
In the long term, I'd like this feature and option to be removed.

Reply:
I saw the discussions in earlier PR's. Will think about it, perhaps @elupus already has some ideas.

Reply:
@MartinHjelmare @elupus What if we remove it and refer in the docs to `template`? I think this feature is only really needed when a user wants to have meaningful value in the UI (other things can be done through state events or checking last_updated). The only real downside is that sensor.time has 1 minute accuracy.

```
binary_sensor:
  - platform: template
    sensors:
      bedroom_occupancy:
        friendly_name: Test Off Delay
        value_template: >-
          {% if (as_timestamp(now()) - as_timestamp(states.binary_sensor.ac_213c7f2_48.last_updated) < 10) %}
            {{ is_state('binary_sensor.ac_213c7f2_48', 'On') }}
          {% else %}
            {{ "false" }}
          {% endif %}
        entity_id: sensor.time

sensor:
  - platform: time_date
    display_options:
      - "time"
```

Reply:
The template sensor support delay_off itself....

Reply:
The way I read the docs, it does something different. I read it as a delay after an off state has been set that the state really goes to off, but it needs someone to tell off before really going off.